java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java

```diff
@@ -115,10 +115,13 @@ public Tree syntaxTree() {
     public Node parent;
     private final List<LearnedConstraint> learnedConstraints;
 
+    private final List<LearnedValue> learnedSymbols;
+
     Node(ProgramPoint programPoint, @Nullable ProgramState programState) {
       this.programPoint = programPoint;
       this.programState = programState;
       learnedConstraints = new ArrayList<>();
+      learnedSymbols = new ArrayList<>();
```

Comment:
In this very case I don't think this should even be debated given the general size of the list.
In those case I tend to stick with arrayList : http://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist